### PR TITLE
Add bm60hsrgb_poker/rev2, a continuation of #15001

### PR DESCRIPTION
**Pull request by peepeetee**
After a discussion with @bdtc123, we decided to have me take over #15001, as his internet connection is spotty and I have too much time on my hands :P

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [x] Documentation

## Issues Fixed or Closed by This PR

* #14964

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/kprepublic/bm60hsrgb_poker/rev2/config.h [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+
+#    define RGB_DISABLE_AFTER_TIMEOUT 0 // number of ticks to wait until disabling effects
+#    define RGB_DISABLE_WHEN_USB_SUSPENDED true // turn off effects when suspended
+#    define RGB_MATRIX_KEYPRESSES
+#    define DISABLE_RGB_MATRIX_GRADIENT_UP_DOWN
+#    define DISABLE_RGB_MATRIX_BAND_SAT
+#    define DISABLE_RGB_MATRIX_BAND_PINWHEEL_SAT
+#    define DISABLE_RGB_MATRIX_BAND_SPIRAL_SAT
+#    define DISABLE_RGB_MATRIX_RAINDROPS
+#    define DISABLE_RGB_MATRIX_JELLYBEAN_RAINDROPS
```
Comment (drashna) [inline keyboards/kprepublic/bm60hsrgb_poker/rev2/config.h:81]: Just a heads up, RGB matrix no longer enables all of the RGB modes, and now require explicit enabling of modes. 

```c
#define ENABLE_RGB_MATRIX_ALPHAS_MODS
#define ENABLE_RGB_MATRIX_GRADIENT_UP_DOWN
#define ENABLE_RGB_MATRIX_GRADIENT_LEFT_RIGHT
#define ENABLE_RGB_MATRIX_BREATHING
#define ENABLE_RGB_MATRIX_BAND_SAT
#define ENABLE_RGB_MATRIX_BAND_VAL
#define ENABLE_RGB_MATRIX_BAND_PINWHEEL_SAT
#define ENABLE_RGB_MATRIX_BAND_PINWHEEL_VAL
#define ENABLE_RGB_MATRIX_BAND_SPIRAL_SAT
#define ENABLE_RGB_MATRIX_BAND_SPIRAL_VAL
#define ENABLE_RGB_MATRIX_CYCLE_ALL
#define ENABLE_RGB_MATRIX_CYCLE_LEFT_RIGHT
#define ENABLE_RGB_MATRIX_CYCLE_UP_DOWN
#define ENABLE_RGB_MATRIX_RAINBOW_MOVING_CHEVRON
#define ENABLE_RGB_MATRIX_CYCLE_OUT_IN
#define ENABLE_RGB_MATRIX_CYCLE_OUT_IN_DUAL
#define ENABLE_RGB_MATRIX_CYCLE_PINWHEEL
#define ENABLE_RGB_MATRIX_CYCLE_SPIRAL
#define ENABLE_RGB_MATRIX_DUAL_BEACON
#define ENABLE_RGB_MATRIX_RAINBOW_BEACON
#define ENABLE_RGB_MATRIX_RAINBOW_PINWHEELS
#define ENABLE_RGB_MATRIX_RAINDROPS
#define ENABLE_RGB_MATRIX_JELLYBEAN_RAINDROPS
#define ENABLE_RGB_MATRIX_HUE_BREATHING
#define ENABLE_RGB_MATRIX_HUE_PENDULUM
#define ENABLE_RGB_MATRIX_HUE_WAVE
#define ENABLE_RGB_MATRIX_PIXEL_FRACTAL
#define ENABLE_RGB_MATRIX_PIXEL_RAIN

#define ENABLE_RGB_MATRIX_TYPING_HEATMAP
#define ENABLE_RGB_MATRIX_DIGITAL_RAIN

#define ENABLE_RGB_MATRIX_SOLID_REACTIVE_SIMPLE
#define ENABLE_RGB_MATRIX_SOLID_REACTIVE
#define ENABLE_RGB_MATRIX_SOLID_REACTIVE_WIDE
#define ENABLE_RGB_MATRIX_SOLID_REACTIVE_MULTIWIDE
#define ENABLE_RGB_MATRIX_SOLID_REACTIVE_CROSS
#define ENABLE_RGB_MATRIX_SOLID_REACTIVE_MULTICROSS
#define ENABLE_RGB_MATRIX_SOLID_REACTIVE_NEXUS
#define ENABLE_RGB_MATRIX_SOLID_REACTIVE_MULTINEXUS
#define ENABLE_RGB_MATRIX_SPLASH
#define ENABLE_RGB_MATRIX_MULTISPLASH
#define ENABLE_RGB_MATRIX_SOLID_SPLASH
#define ENABLE_RGB_MATRIX_SOLID_MULTISPLASH
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]

File: keyboards/kprepublic/bm60hsrgb_poker/rev2/rules.mk [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+LTO_ENABLE = yes
+
+# RGB Matrix is required to support per-key LEDs connected to IS31FL3733.
+RGB_MATRIX_ENABLE = yes
+
+# The simpler RGB Lighting code can be enabled to control just the underglow
+# LEDs from it; leave it disabled to make the RGB Matrix code control all LEDs,
+# including underglow.
+RGBLIGHT_ENABLE = no
+
```
Comment (fauxpark) [inline keyboards/kprepublic/bm60hsrgb_poker/rev2/rules.mk:28]:
```suggestion
```

File: keyboards/kprepublic/bm60hsrgb_poker/rev2/rev2.h [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    { K10,   K11,   K12,   K13,   K14,   K15,   K16,   K17,   K18,   K19,   K1A,   K1B,   K1C,   K1D   }, \
+    { K20,   K21,   K22,   K23,   K24,   K25,   K26,   K27,   K28,   K29,   K2A,   K2B,   KC_NO, K2D   }, \
+    { K30,   K31,   K32,   K33,   K34,   K35,   K36,   K37,   K38,   K39,   K3A,   KC_NO, KC_NO, K3D   }, \
+    { K40,   K41,   K42,   KC_NO, KC_NO, K45,   KC_NO, KC_NO, KC_NO, K49,   K4A,   K4B,   KC_NO, K4D   }, \
+}
+
+
+
+
+
```
Comment (fauxpark) [inline keyboards/kprepublic/bm60hsrgb_poker/rev2/rev2.h:37]:
```suggestion
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/kprepublic/bm60hsrgb_poker/readme.md [hunk 4]
```diff
@@ -2,12 +2,18 @@
...
+    for rev1: make kprepublic/bm60hsrgb_poker/rev1:default
+    for rev2: make kprepublic/bm60hsrgb_poker/rev2:default
+
+
+To determine your board revision, check the PCB. For boards with bm60v2_poker printed on the PCB, use rev2. Otherwise, use rev1
 
 See the [build environment setup](https://docs.qmk.fm/#/getting_started_build_tools) and the [make instructions](https://docs.qmk.fm/#/getting_started_make_guide) for more information. Brand new to QMK? Start with our [Complete Newbs Guide](https://docs.qmk.fm/#/newbs).
+
+
+
```
Comment (fauxpark) [inline keyboards/kprepublic/bm60hsrgb_poker/readme.md:19]:
```suggestion
```

File: keyboards/kprepublic/bm60hsrgb_poker/readme.md [hunk 5]
```diff
@@ -2,12 +2,18 @@
...
+* Hardware Supported: BM60HSRGB_Poker/rev2
 * Hardware Availability: [KP Republic](https://kprepublic.com/products/bm60-rgb-poker-60-gh60-hot-swap-custom-mechanical-keyboard-pcb-program-qmk-underglow-type-c)
 
 Make example for this keyboard (after setting up your build environment):
 
-    make kprepublic/bm60hsrgb_poker/rev1:default
+    for rev1: make kprepublic/bm60hsrgb_poker/rev1:default
+    for rev2: make kprepublic/bm60hsrgb_poker/rev2:default
+
+
```
Comment (fauxpark) [inline keyboards/kprepublic/bm60hsrgb_poker/readme.md:13]:
```suggestion
```

File: keyboards/kprepublic/bm60hsrgb_poker/rev2/config.h [hunk 6]
```diff
@@ -0,0 +1,157 @@
...
+#define ENABLE_RGB_MATRIX_SOLID_SPLASH
+#define ENABLE_RGB_MATRIX_SOLID_MULTISPLASH
+
+
+
+#define RGB_MATRIX_MAXIMUM_BRIGHTNESS 180
+
+
+
+
```
Comment (fauxpark) [inline keyboards/kprepublic/bm60hsrgb_poker/rev2/config.h:127]:
```suggestion
#define RGB_MATRIX_MAXIMUM_BRIGHTNESS 180
```

File: keyboards/kprepublic/bm60hsrgb_poker/rev2/config.h [hunk 7]
```diff
@@ -0,0 +1,157 @@
...
+#endif
+
+#define DRIVER_LED_TOTAL (ISSI_LED_TOTAL + WS2812_LED_TOTAL)
+
+#ifdef RGB_MATRIX_ENABLE
+#    define RGB_MATRIX_KEYPRESSES
+#    define RGB_MATRIX_FRAMEBUFFER_EFFECTS
+#endif
+
+
```
Comment (fauxpark) [inline keyboards/kprepublic/bm60hsrgb_poker/rev2/config.h:157]:
```suggestion
```

File: keyboards/kprepublic/bm60hsrgb_poker/rev2/rev2.c [hunk 8]
```diff
@@ -0,0 +1,204 @@
...
+
+#ifdef RGB_MATRIX_ENABLE
+
+#    include "i2c_master.h"
+#    include "is31fl3733.h"
+#    include "ws2812.h"
+
+
+
+
```
Comment (fauxpark) [inline keyboards/kprepublic/bm60hsrgb_poker/rev2/rev2.c:26]:
```suggestion
```

File: keyboards/kprepublic/bm60hsrgb_poker/rev2/rev2.c [hunk 9]
```diff
@@ -0,0 +1,204 @@
...
+__attribute__ ((weak))
+void rgb_matrix_indicators_user(void) {
+    if (host_keyboard_led_state().caps_lock) {
+        rgb_matrix_set_color(28, 0xFF, 0xFF, 0xFF);
+    }
+}
+
+
+
```
Comment (fauxpark) [inline keyboards/kprepublic/bm60hsrgb_poker/rev2/rev2.c:141]:
```suggestion
```

File: keyboards/kprepublic/bm60hsrgb_poker/rev2/rev2.c [hunk 10]
```diff
@@ -0,0 +1,204 @@
...
+    .init          = rgb_matrix_driver_init,
+    .flush         = rgb_matrix_driver_flush,
+    .set_color     = rgb_matrix_driver_set_color,
+    .set_color_all = rgb_matrix_driver_set_color_all,
+};
+// clang-format on
+
+#endif /* RGB_MATRIX_ENABLE */
+
+
```
Comment (fauxpark) [inline keyboards/kprepublic/bm60hsrgb_poker/rev2/rev2.c:204]:
```suggestion
```

File: keyboards/kprepublic/bm60hsrgb_poker/rev2/rules.mk [hunk 11]
```diff
@@ -0,0 +1,33 @@
...
+# The custom RGB Matrix driver combines IS31FL3733 and WS2812; things that are
+# normally done by common_features.mk for both of these drivers need to be done
+# here manually.
+RGB_MATRIX_DRIVER = custom
+COMMON_VPATH += $(DRIVER_PATH)/led/issi
+SRC += is31fl3733.c
+QUANTUM_LIB_SRC += i2c_master.c
+WS2812_DRIVER_REQUIRED = yes
+
+
```
Comment (fauxpark) [inline keyboards/kprepublic/bm60hsrgb_poker/rev2/rules.mk:33]:
```suggestion
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by fauxpark — APPROVED** [review]

**Pull request merged**